test/integration/pods/pods_test.go

```diff
@@ -662,3 +663,60 @@ func TestPodUpdateEphemeralContainers(t *testing.T) {
 		integration.DeletePodOrErrorf(t, client, ns.Name, pod.Name)
 	}
 }
+
+// TestPodEphemeralContainersDisabled tests that the API server returns a 404 when the feature is disabled (because the subresource won't exist).
+// This validates that the feature gate is working, but kubectl also uses the 404 to guess that the feature is disabled on the server.
+func TestPodEphemeralContainersDisabled(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.EphemeralContainers, false)()
```

Comment:
I wonder why is this being deferred, should not it be called directly instead?

Reply:
It's deferring execution of the return value rather than `SetFeatureGateDuringTest()` itself. The first call sets the feature to false and the second call restores its previous value.